rust/chains/hyperlane-ethereum/src/provider.rs

```diff
@@ -46,30 +46,29 @@ where
     M: Middleware + 'static,
 {
     #[instrument(err, skip(self))]
-    async fn get_block_by_hash(&self, hash: &H256) -> eyre::Result<BlockInfo> {
+    async fn get_block_by_hash(&self, hash: &H256) -> ChainResult<BlockInfo> {
         let block = get_with_retry_on_none(|| self.provider.get_block(*hash)).await?;
         Ok(BlockInfo {
             hash: *hash,
             timestamp: block.timestamp.as_u64(),
             number: block
                 .number
-                .ok_or_else(|| eyre!("Block is not part of the chain yet {}", hash))?
+                .ok_or(HyperlaneProviderError::BlockIsNotPartOfChainYet(*hash))?
```

Comment:
Not a big deal but I wonder if this should be like "UnknownBlockHash" or something? The assumption that the block isn't a part of the chain yet is predicated on the caller supplying a hash that it knows will exist but doesn't yet. Imo the function should avoid making assumptions of how it's called

Reply:
Yeah, that is a good point. Fixing.

Reply:
Oh, actually, no, this is only if the block hash is known but the block number is not set. Which according to the docs is `None if pending`.